lib/nerves_hub_link_common/update_manager.ex

```diff
@@ -183,6 +183,11 @@ defmodule NervesHubLinkCommon.UpdateManager do
 
   @spec maybe_update_firmware(UpdateAvailable.t(), State.t()) ::
           State.download_started() | State.download_rescheduled() | State.t()
+  defp maybe_update_firmware(%UpdateAvailable{update_available: false}, %State{} = state) do
```

Comment:
Should it also bail early if the `:update_available` key doesn't exist? Isn't that a possible case prior to your NH fix? Like would this affect people running NH servers and who aren't syncing with upstream regularly?